src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/PasswordChangeRequest.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.Handlers
+{
+    internal class PasswordChangeRequest
+    {
+        public string NewPassword { get; internal set; }
+
+        public SqlCredential Credential { get; internal set; }
+
+        public SecureString NewSecurePassword { get; internal set; }
```

Comment:
This corresponds to a public API surface, but SecureString [isn't actively recommended](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-security-securestring) any more. We'll have to decrypt the SecureString and (merely) obfuscate it in the packet buffer anyway, so it'll be stored in memory whatever happens. Depending on its security implications, we might be able to remove this field, perform the transformation from SecureString to string in PasswordChangeRequest and thus remove the need to translate this in the login handler.

Reply:
Yeah, but out of scope of this PR. We will need to change the public API and then we can remove all this occurrences. 

I have brought in the consideration of SecureString, but haven't implemented it anywhere yet. 

There are some clever techniques that are used in the current SqlClient implementation to convert SecureString as late as possible to I/O buffer, but yes its not foolproof.